### PR TITLE
add softfloat3e library to Bochs to replace old softfloat2 (to get fp…

### PR DESCRIPTION
**Pull request by stlintel**
…16 support)

with a lot of my updates, cleanups and extensions for x86 features like denormal and/or undeflow/overflow handling

### CODE REVIEW

File: bochs/Makefile.in [hunk 1]
```diff
@@ -349,6 +350,11 @@ cpu/fpu/libfpu.a::
 	$(MAKE) $(MDEFINES) libfpu.a
 	@CD_UP_TWO@
 
+cpu/softftloat3e/libsoftftloat.a::
```
Comment (Vort) [inline bochs/Makefile.in:353]: should be `cpu/softfloat3e/libsoftfloat.a::`
(`ftft` breaks build)

Comment (stlintel) [reply bochs/Makefile.in]: Thanks !

**Comment by Vort** [comment]
Latest commit seems to break many things (Quake for example).

**Comment by Vort** [comment]
Also I noticed warning from compiler. Maybe false alarm, do not know, but showing it anyway:
```
avx512_rcp14.cc:8325:28: warning: logical not is only applied to the left hand side of this bitwise operator [-Wlogical-not-parentheses]
  fraction32 &= ~(Bit16u) (! (roundBits ^ 0x20) & 0x1);
```

**Comment by stlintel** [comment]
This is just redundant AND. Will fix to silence the warning.

**Comment by stlintel** [comment]
> Latest commit seems to break many things (Quake for example).

For some weird reason quite naive change destroyed everything. on first instruction after reset all FPU appear as initialized to some weird denormals instead of 0 which should be initialized with memset. smth was wrong in compiler with moving it from softfloat.h to softfloat3 codebase.  will take on it later but first I want to finish convertion and cleanup for everything f32/f64 and might be even push it to master. Of course better after coming release.

**Comment by Vort** [comment]
> on first instruction after reset all FPU appear as initialized to some weird denormals instead of 0

It happens because of wrong `LITTLEENDIAN` definition.
Can be hackfixed like this: https://github.com/Vort/Bochs/commit/9b01ef324cebdb217c4844e23190c2009665f058.

**Pull request merged**